src/services/transaction.ts

```diff
@@ -23,13 +23,21 @@ async function cancel(id: string) {
 
   return res;
 }
-
-async function list(apiKey: string): Promise<IScheduledForUser[]> {
+// Promise<IScheduledForUser[]>
+async function list(apiKey: string, opts: any): Promise<[IScheduledForUser[], number]> {
   const user = await UserService.validateApiKey(apiKey);
   const scheduleds = await Scheduled.find({ userId: user.id }).exec();
```

Comment:
this is where the filtering / sorting should be applied now, so we don't fetch / process all the data from db on each reqeust